### ISSUE TITLE
some confusion about the g_adv_loss in the code

### ISSUE THREAD

**Issue by 973891422** [ronny3050/AdvFaces#5]
In the code:
`g_adv_loss = tf.reduce_mean(
                    tf.nn.sigmoid_cross_entropy_with_logits(
                        logits=self.D_fake, labels=tf.ones_like(self.D_fake)
                    )
                )`
The  tf.nn.sigmoid_cross_entropy_with_logits has arg labels = 1
Does this mean L_g_gan = log(D(x_adv)) instead of L_g_gan = log(1-D(x_adv))？In the paper the L_g_gan = E(log(1-D(x_adv))), which confuses me

**Comment by 973891422** [comment]
Sorry, I've found the answer in the paper _Generative Adversarial Nets_

>  Early in learning,when G is poor, D can reject samples with high confidence because they are clearly different from the training data.  In this case, log(1−D(G(z))) saturates.  Rather than training G to minimize log(1−D(G(z))) we can train G to maximize logD(G(z)).
